ibm/service/power/resource_ibm_pi_network.go

```diff
@@ -272,17 +271,22 @@ func resourceIBMPINetworkUpdate(ctx context.Context, d *schema.ResourceData, met
 		return diag.FromErr(err)
 	}
 
-	if d.HasChanges(helpers.PINetworkName, helpers.PINetworkDNS, helpers.PINetworkIPAddressRange) {
+	if d.HasChanges(helpers.PINetworkName, helpers.PINetworkDNS, helpers.PINetworkGateway, helpers.PINetworkIPAddressRange) {
 		networkC := st.NewIBMPINetworkClient(ctx, sess, cloudInstanceID)
 		body := &models.NetworkUpdate{
 			DNSServers: flex.ExpandStringList((d.Get(helpers.PINetworkDNS).(*schema.Set)).List()),
 		}
 		networkType := d.Get(helpers.PINetworkType).(string)
-		if d.HasChange(helpers.PINetworkIPAddressRange) {
+		if d.HasChange(helpers.PINetworkIPAddressRange) || d.HasChange(helpers.PINetworkGateway) {
 			if networkType == Vlan {
-				body.IPAddressRanges = getIPAddressRanges(d.Get(helpers.PINetworkIPAddressRange).([]interface{}))
+				if d.HasChange(helpers.PINetworkIPAddressRange) {
+					body.IPAddressRanges = getIPAddressRanges(d.Get(helpers.PINetworkIPAddressRange).([]interface{}))
+				}
+				if d.HasChange(helpers.PINetworkGateway) {
+					body.Gateway = flex.PtrToString(d.Get(helpers.PINetworkGateway).(string))
```

Comment:
Why using PtrToString?

Reply:
body.Gateway is *string and PtrToString takes a string and outputs a *string. I think the flex function might be named backwards.